sway/sway_text_node.c

```diff
@@ -87,7 +87,7 @@ static void render_backing_buffer(struct text_buffer *buffer) {
 	}
 
 	float scale = buffer->scale;
-	int width = ceil(buffer->props.width * scale);
+	int width = ceil(get_text_width(&buffer->props) * scale);
```

Comment:
The whole text was rendered to avoid rendering again when resizing a window.

I believe this change breaks `sway_text_node_set_max_width()`.

Reply:
Oh, but we call `render_backing_buffer()` in `sway_text_node_set_max_width()`. According to git log it seems like it was there since the beginning.

This begs the question: do we need `update_source_box()` for anything?

Reply:
Yeah I also thought that the source box was to optimize resize, but as you mention it's rendered ineffective.

It makes sense to do, but the question is if we want to fix that now or just whack it for now and try again later. From the perspective of fixing the crash I'm leaning towards the latter.

Reply:
Yeah, makes sense to fix now and leave optimizations for later.